### ISSUE TITLE
Add haptic feedback for Switch toggles

### ISSUE THREAD

**Issue by jklint-g** [flutter/flutter#20606]
Toggling a native boolean switch control on iOS gives short haptic feedback on newer phones. The Flutter Switch widget should match this.

**Comment by jklint-g** [comment]
Note that there's a pretty extensive set of system haptics on iOS which should be evaluated for Flutter https://www.macstories.net/stories/system-haptics-on-the-iphone-7/
